docs/integrations/nestjs.md

```diff
@@ -0,0 +1,287 @@
+---
+sidebar_position: 2
+title: Nest.js
+---
+import InstallNode from '/docs/partials/_install_node.mdx';
+
+This guide shows you how to integrate [DBOS Transact](https://github.com/dbos-inc/dbos-transact) with your existing [Nest.js](https://nestjs.com/) application.
+
+Nest.js uses Dependency Injection (an [inversion of control](https://en.wikipedia.org/wiki/Inversion_of_control) technique) and encourages you to write your business logic in services that can be injected into your controllers or other services. Such injectable services are exported by Nest.js modules as "providers".
+
+In this guide, we will show you how to make Nest.js services reliable with DBOS Transact and export them as Nest.js providers.
+
+## DBOS-ify your Nest.js application
+
+### 1. Installation and requirements
+
+:::info
+The example in this guide is based of a new Nest.js application, initialized with `nest new nest-starter` and configured to use [NPM](https://www.npmjs.com/).
+:::
+
+First, install the DBOS typecript SDK with `npm install @dbos-inc/dbos-sdk`.
+
+DBOS requires a postgres database. Add a `dbos-config.yaml` file to the root of your project with your database connection information:
+```yaml
+database:
+  hostname: localhost
+  port: 5432
+  username: ${PGUSER}
+  password: ${PGPASSWORD}
+  app_db_client: knex
+```
+
+### 2. Bootstraping DBOS
+
+Modify your bootstrap function to import the DBOS SDK and launch DBOS Transact:
+
+```typescript
+import { NestFactory } from '@nestjs/core';
+import { AppModule } from './app.module';
+// highlight-next-line
+import { DBOS } from "@dbos-inc/dbos-sdk";
+
+async function bootstrap() {
+  const app = await NestFactory.create(AppModule);
+  // highlight-next-line
+  await DBOS.launch();
+  await app.listen(process.env.PORT ?? 3000);
+}
+bootstrap();
+```
+
+Optionally, you can pass a `nestApp` to `DBOS.launch()` to install an [OpenTelemetry](https://opentelemetry.io/) tracing middleware to your app:
+```typescript
+await DBOS.launch({ nestApp: app });
+```
+DBOS natively generates OTel traces for your workflows. This middleware wires them with the traces context of external requests.
+
+### 3. DBOS-ify your services
+
+For this demonstration, we will use a simple service that does two things: sending an HTTP request and inserting a record in the database. You can find the full code for this example on [github](TODO).
```

Comment:
Condense the example, make it prescriptive instead of example-driven, and explain Nest.js less and DBOS (particularly configured instances) slightly more.